src/components/Attachments/SelectAttachment/SelectAttachment.tsx

```diff
@@ -0,0 +1,167 @@
+// Copyright (C) TOSHIBA CORPORATION, 2023. Part of the SW360 Frontend Project.
+// Copyright (C) Toshiba Software Development (Vietnam) Co., Ltd., 2023. Part of the SW360 Frontend Project.
+
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+
+// SPDX-License-Identifier: EPL-2.0
+// License-Filename: LICENSE
+
+'use client'
+import Modal from 'react-bootstrap/Modal'
+import Button from 'react-bootstrap/Button'
+import { Session } from '@/object-types/Session'
+import React, { useState, useRef } from 'react'
+import { useTranslations } from 'next-intl'
+import { COMMON_NAMESPACE } from '@/object-types/Constants'
+import AttachmentDetail from '@/object-types/AttachmentDetail'
+import { SW360_API_URL } from '@/utils/env'
+import ComponentPayload from '@/object-types/ComponentPayLoad'
+import styles from './SelectAttachment.module.css'
+
+interface Props {
+    show: boolean
+    setShow: React.Dispatch<React.SetStateAction<boolean>>
+    session: Session
+    attachmentUpload: any
+    setAttachmentFromUpload: any
+    onReRender: () => void
+    componentData: ComponentPayload
+    setComponentData: React.Dispatch<React.SetStateAction<ComponentPayload>>
+}
+
+const SelectAttachment = ({
+    show,
+    setShow,
+    session,
+    attachmentUpload,
+    setAttachmentFromUpload,
+    onReRender,
+    componentData,
+    setComponentData,
+}: Props) => {
+    const t = useTranslations(COMMON_NAMESPACE)
+    const [files, setFiles] = useState([])
+    const inputRef = useRef(null)
+
+    const handleFileChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const files = Array.prototype.slice.call(e.target.files)
+        console.log(files)
+        if (files.length > 0) {
```

Comment:
`
 if (files.length > 0) {
            setFiles(files)
        } else {
            setFiles([])
        }
`
to
`setFiles(files)`